crates/biome_fs/src/fs/os.rs

```diff
@@ -513,6 +513,12 @@ impl TemporaryFs {
             fs::remove_dir_all(path.as_path()).unwrap();
         }
         fs::create_dir(&path).unwrap();
+
+        // On macOS, the temporary directory is in `/var`, which is a symlink to `/private/var`.
+        // We need to get the actual path here, or we will get path inconsistency.
+        #[cfg(target_os = "macos")]
```

Comment:
I wanted to do `std::fs::canonicalize` for all platforms here, but it will change the path to `\\?\C:\...` on Windows, which is not properly handled in this project. 😢